examples/remarketing/add_merchant_center_dynamic_remarketing_campaign.py

```diff
@@ -0,0 +1,358 @@
+    ).DISPLAY
+    campaign.status = client.get_type("CampaignStatusEnum", version="v5").PAUSED
+    campaign.campaign_budget = client.get_service(
+        "CampaignBudgetService", version="v5"
+    ).campaign_budget_path(customer_id, campaign_budget_id)
+    campaign.manual_cpc.CopyFrom(client.get_type("ManualCpc", version="v5"))
+
+    # Issues a mutate request to add the campaign.
+    campaign_response = campaign_service.mutate_campaigns(
+        customer_id, [campaign_operation]
+    )
+    campaign_resource_name = campaign_response.results[0].resource_name
+    print(f"Created campaign with resource name '{campaign_resource_name}'.")
+
+    return campaign_resource_name
+
+
+def _create_ad_group(client, customer_id, campaign_resource_name):
+    """Creates an ad group for the remarketing campaign.
+
+    Args:
+        client: An initialized GoogleAds client.
+        customer_id: The Google Ads customer ID.
+        campaign_resource_name: The resource name of the target campaign.
+    Returns:
+        The string resource name of the newly created ad group.
+    """
+    # Gets the AdGroupService.
+    ad_group_service = client.get_service("AdGroupService", version="v5")
+
+    # Creates an ad group operation and configures the new ad group.
+    ad_group_operation = client.get_type("AdGroupOperation", version="v5")
+    ad_group = ad_group_operation.create
+    ad_group.name = "Dynamic remarketing ad group"
+    ad_group.campaign = campaign_resource_name
+    ad_group.status = client.get_type("AdGroupStatusEnum", version="v5").ENABLED
+
+    # Issues a mutate request to add the ad group.
+    ad_group_response = ad_group_service.mutate_ad_groups(
+        customer_id, [ad_group_operation]
+    )
+    ad_group_resource_name = ad_group_response.results[0].resource_name
+
+    return ad_group_resource_name
+
+
+def _create_ad(client, customer_id, ad_group_resource_name):
+    """Creates the responsive display ad.
+
+    Args:
+        client: An initialized GoogleAds client.
+        customer_id: The Google Ads customer ID.
+        ad_group_resource_name: The resource name of the target ad group.
+    """
+    # Get the AdGroupAdService client.
+    ad_group_ad_service = client.get_service("AdGroupAdService", version="v5")
+
+    # Upload image assets for the ad.
+    marketing_image_resource_name = _upload_image_asset(
+        client, customer_id, "https://goo.gl/3b9Wfh", "Marketing Image"
+    )
+    square_marketing_image_resource_name = _upload_image_asset(
+        client, customer_id, "https://goo.gl/mtt54n", "Square Marketing Image"
+    )
+
+    # Create the relevant asset objects for the ad.
+    marketing_image = client.get_type("AdImageAsset", version="v5")
+    marketing_image.asset = marketing_image_resource_name
+    square_marketing_image = client.get_type("AdImageAsset", version="v5")
+    square_marketing_image.asset = square_marketing_image_resource_name
+    headline = client.get_type("AdTextAsset", version="v5")
```

Comment:
Any reason why the headline and descriptions aren't with the rest of the attribute settings (currently line 202)?

Reply:
Purely readability. My thinking was that all the "Assets" be set up in one block, then all the attributes for the ResponsiveDisplayAdInfo be assigned in one block. To me it seemed more readable to group the setups and the assignments, especially given that there are so many assignments for the ResponsiveDisplayAdInfo.